tools/check_spelling_pedantic.py

```diff
@@ -0,0 +1,344 @@
+#! /usr/bin/env python
+
+from __future__ import print_function
```

Comment:
This is pretty epic. Should we run this as part of the check format job?

Reply:
I'm not quite there yet, for a couple of reasons. First I want to see how well I've done ignoring non-words (all the regexes dealing with hexadecimal/UUIDs/base64 etc). Second, there probably ought to be a fix option for the script, but it'll need to be interactive (because technical jargon).

Reply:
OK, well, this is awesome. You might need to merge master to get a clang-tidy fix. Once this passes tests let's do it!